File: src/main/java/org/usfirst/frc/team1458/lib/core/AutoModeHolder.java
package org.usfirst.frc.team1458.lib.core;

import java.util.ArrayList;

/**
 * Allows switching between multiple AutoModes
 *
 * @author asinghani
 */
public interface AutoModeHolder {
	ArrayList<? extends AutoMode> getAutoModes();

	void setSelectedAutoModeIndex(int index);

	int getSelectedAutoModeIndex();
}

File: src/main/java/org/usfirst/frc/team1458/lib/core/AutoModeHolder.kt
package org.usfirst.frc.team1458.lib.core

import java.util.ArrayList

/**
 * Allows switching between multiple AutoModes
 *
 * @author asinghani
 */
interface AutoModeHolder {
    val autoModes: ArrayList<AutoMode>
        get

    var selectedAutoModeIndex: Int

    val selectedAutoMode: AutoMode
        get() = autoModes[selectedAutoModeIndex]
}